homeassistant/components/media_player/mpd.py

```diff
@@ -202,12 +201,21 @@ def media_album_name(self):
     @property
     def volume_level(self):
         """Return the volume level."""
-        return int(self._status['volume'])/100
+        if 'volume' in self._status:
+            return int(self._status['volume'])/100
 
     @property
     def supported_features(self):
         """Flag media player features that are supported."""
-        return SUPPORT_MPD
+        features = SUPPORT_MPD
+        if not self._status:
+            self.update()
+        if 'volume' in self._status:
```

Comment:
Configuration can easily change if mpd restarts, but I don't believe this level of configuration change can happen at runtime. And since supported_features gets called before the connection to the service is established, there's no good way to interrogate for feature availability.

It looks like bluesound also has to figure out at runtime whether a set of features are available, and just returns None when not yet connected; do you think I should just model after that? ie. instead of forcing at update here, I'll just return None until we have an active connection, at which point I can interrogate the server for capabilities?